extension/jaegerremotesampling/internal/remote_strategy_store.go

```diff
@@ -0,0 +1,59 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package internal // import "github.com/open-telemetry/opentelemetry-collector-contrib/extension/jaegerremotesampling/internal"
+
+import (
+	"context"
+	"fmt"
+
+	grpcstore "github.com/jaegertracing/jaeger/cmd/agent/app/configmanager/grpc"
+	"github.com/jaegertracing/jaeger/cmd/collector/app/sampling/strategystore"
+	"github.com/jaegertracing/jaeger/thrift-gen/sampling"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config/configgrpc"
+	"go.opentelemetry.io/collector/config/configopaque"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+)
+
+type grpcRemoteStrategyStore struct {
+	conn            *grpc.ClientConn
+	delegate        *grpcstore.SamplingManager
+	headerAdditions map[string]configopaque.String
+}
+
+// NewRemoteStrategyStore returns a StrategyStore that delegates to the configured Jaeger gRPC endpoint, making
+// extension-configured enhancements (header additions only for now) to the gRPC context of every outbound gRPC call.
+// Note: it would be nice to expand the configuration surface to include an optional TTL-based caching behavior
+// for service-specific outbound GetSamplingStrategy calls.
+func NewRemoteStrategyStore(
+	ctx context.Context,
+	host component.Host,
+	telemetry component.TelemetrySettings,
+	grpcClientSettings *configgrpc.GRPCClientSettings,
+) (strategystore.StrategyStore, func() error, error) {
+	conn, err := grpcClientSettings.ToClientConn(ctx, host, telemetry)
```

Comment:
We pass all the parameter to create a `*grpc.ClientConn`, right? `conn` is then used to create the delegate and the `conn.Close` is returned as a closer. Also its stored in the structure, but never used.
Wouldnt it be simpler to just pass a `*grpc.ClientConn`? That would remove the closer function as a return type as well as the need to pass a context, component.Host, and component.TelemetrySettings.

Reply:
yeah, I think that's reasonable – no need to "push" all this stuff from inside the previous caller in `extension.go`

I'll move it now

Reply:
(done)

Reply:
this is a lot cleaner, ty